src/vm/prestub.cpp

```diff
@@ -441,6 +441,10 @@ PCODE MethodDesc::MakeJitWorker(COR_ILMETHOD_DECODER* ILHeader, DWORD flags, DWO
                     COR_ILMETHOD *pilHeader = GetILHeader(TRUE);
                     new (ILHeader) COR_ILMETHOD_DECODER(pilHeader, GetMDImport(), NULL);
                 }
+                else
+                {
+                    g_profControlBlock.pProfInterface->JITCompilationStartedNoMetadata((FunctionID) this);
```

Comment:
I didn't notice any check that verifies the profiler implements ICorProfilerCallback8 in this callback path? If you haven't tested with a profiler that does not implement ICorProfilerInfo8/callback8 I think you should. I suspect as-is such a profiler would AV.